### ISSUE TITLE
[🐞] QWIK ERROR Error: Code(3) when returning the cloudflare d1's results object from the routeLoader$

### ISSUE THREAD

**Issue by salihbenlalla** [BuilderIO/qwik#3449]
### Which component is affected?

Qwik City (routing)

### Describe the bug

i have this code

```ts
import { component$ } from "@builder.io/qwik";
import {
  routeLoader$,
  type RequestEventLoader,
} from "@builder.io/qwik-city";
import { type PlatformCloudflarePages } from "@builder.io/qwik-city/middleware/cloudflare-pages";
import type { D1Database, D1Result } from "@miniflare/d1";

declare module "@builder.io/qwik-city/middleware/cloudflare-pages" {
  interface PlatformCloudflarePages {
    DB: D1Database;
  }
}

interface ListItem {
  id: number;
  text: string;
}

export const useListLoader = routeLoader$(
  async (ev: RequestEventLoader<PlatformCloudflarePages>) => {
    if (ev.platform.DB) {
      const { results } = (await ev.platform.DB.prepare(
        "SELECT * FROM Todos"
      ).all()) as D1Result<ListItem>;

      const list = results ?? [];

      return list;
    }
    return [];
  }
);

export default component$(() => {
  const list = useListLoader();

  return (
    <>
        <div>
          {(list.value.length && (
            <ul>
              {list.value.map((item, index) => (
                <li key={`items-${index}`}>{item.text}</li>
              ))}
            </ul>
          )) || <span>No items found</span>}
        </div>
    </>
  );
});
```

when i `npm run build` => `npm run deploy` and open the page in localhost i get this error

```bash
QWIK ERROR Error: Code(3)
    at createError (/home/salih/Desktop/qwik-demo/server/assets/@qwik-city-plan-582a5475.js:33:34)
    at logError (/home/salih/Desktop/qwik-demo/server/assets/@qwik-city-plan-582a5475.js:29:52)
    at logErrorAndStop (/home/salih/Desktop/qwik-demo/server/assets/@qwik-city-plan-582a5475.js:34:57)
    at qError (/home/salih/Desktop/qwik-demo/server/assets/@qwik-city-plan-582a5475.js:40:10)
    at /home/salih/Desktop/qwik-demo/server/assets/@qwik-city-plan-582a5475.js:1450:11
    at Array.map (<anonymous>)
    at _pauseFromContexts (/home/salih/Desktop/qwik-demo/server/assets/@qwik-city-plan-582a5475.js:1416:30)
    at beforeClose (/home/salih/Desktop/qwik-demo/server/assets/entry.ssr-e23eca27.js:488:30)
    at /home/salih/Desktop/qwik-demo/server/assets/@qwik-city-plan-582a5475.js:1734:20
    at /home/salih/Desktop/qwik-demo/server/assets/@qwik-city-plan-582a5475.js:1983:21 { id: 1, text: 'take a shower' }
GET /error-code-3/ 200 OK (204.22ms)
Error: Code(3)
    at createError (/home/salih/Desktop/qwik-demo/server/assets/@qwik-city-plan-582a5475.js:33:34)
    at logError (/home/salih/Desktop/qwik-demo/server/assets/@qwik-city-plan-582a5475.js:29:52)
    at logErrorAndStop (/home/salih/Desktop/qwik-demo/server/assets/@qwik-city-plan-582a5475.js:34:57)
    at qError (/home/salih/Desktop/qwik-demo/server/assets/@qwik-city-plan-582a5475.js:40:10)
    at /home/salih/Desktop/qwik-demo/server/assets/@qwik-city-plan-582a5475.js:1450:11
    at Array.map (<anonymous>)
    at _pauseFromContexts (/home/salih/Desktop/qwik-demo/server/assets/@qwik-city-plan-582a5475.js:1416:30)
    at beforeClose (/home/salih/Desktop/qwik-demo/server/assets/entry.ssr-e23eca27.js:488:30)
    at /home/salih/Desktop/qwik-demo/server/assets/@qwik-city-plan-582a5475.js:1734:20
    at /home/salih/Desktop/qwik-demo/server/assets/@qwik-city-plan-582a5475.js:1983:21
```

but when i recreate the `results` object and return it like this:

```ts
export const useListLoader = routeLoader$(
  async (ev: RequestEventLoader<PlatformCloudflarePages>) => {
    if (ev.platform.DB) {
      const { results } = (await ev.platform.DB.prepare(
        "SELECT * FROM Todos"
      ).all()) as D1Result<ListItem>;

      const list = results ?? [];

      return list.map((item) => {
        return { id: item.id, text: item.text };
      });
    }
    return [];
  }
);
```

the mentioned error goes away but a new error appears:

```bash
TypeError: Invalid state: WritableStream is locked
    at new NodeError (node:internal/errors:387:5)
    at WritableStream.close (node:internal/webstreams/writablestream:220:9)
    at /home/salih/Desktop/qwik-demo/server/entry.cloudflare-pages.js:481:26
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at Object.next (/home/salih/Desktop/qwik-demo/server/entry.cloudflare-pages.js:623:9)
    at runNext (/home/salih/Desktop/qwik-demo/server/entry.cloudflare-pages.js:854:5) {
  code: 'ERR_INVALID_STATE'
}
```

i think it's the same as #3269

### Reproduction

https://github.com/salihbenlalla/qwik-demo

### Steps to reproduce

1. clone the repo in the above url
2. run `npm install`
3. run `wrangler d1 create DB` to create a new d1 database
4. copy the credentials of your database and add them to `wrangler.toml` file in your root directory.
```toml
[[ d1_databases ]]
binding = "<BINDING_NAME>"
database_name = "<DATABASE_NAME>"
database_id = "<UUID>"
```
5. run `wrangler d1 execute DB --local --file=./schema.sql` to execute the schema against your database.
6. run `npm run build`
7. run `npm run deploy`
8. go to `http://127.0.0.1:8788/error-code-3/` to see the first error in your terminal.
9. go to `http://127.0.0.1:8788/err-invalid-state/` to see the second error in your terminal.

### System Info

```shell
System:
    OS: Linux 5.10 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
    CPU: (4) x64 Intel(R) Core(TM) i3-2370M CPU @ 2.40GHz
    Memory: 5.19 GB / 7.66 GB
    Container: Yes
    Shell: 3.5.1 - /usr/bin/fish
  Binaries:
    Node: 16.18.0 - ~/.local/share/nvm/v16.18.0/bin/node
    npm: 8.19.2 - ~/.local/share/nvm/v16.18.0/bin/npm
  Browsers:
    Chromium: 111.0.5563.64
  npmPackages:
    @builder.io/qwik: 0.23.0 => 0.23.0 
    @builder.io/qwik-city: 0.6.5 => 0.6.5 
    undici: 5.21.0 => 5.21.0 
    vite: 4.2.0 => 4.2.0
```

**Comment by manucorporat** [comment]
This is too hard for me to reproduce, can you try to debug what's the DB result?
console.log, see if t's a custom class

**Comment by salihbenlalla** [comment]
> This is too hard for me to reproduce, can you try to debug what's the DB result? console.log, see if t's a custom class

It's just an array of object literals:
```js
[
    {
        id: 1,
        text: "take a shower"
    },
    {
        id: 1,
        text: "take a shower"
    }
]
```

And this is what confused me!

it's not an instance of a class or something

**Comment by manucorporat** [comment]
could be, but i doubt it because of the error

**Comment by mpeg** [comment]
I'm not sure about the initial error here, but the WritableStream error is fixed by #5352
